inlang/source-code/paraglide/paraglide-js-adapter-unplugin/src/index.ts

```diff
@@ -59,9 +61,18 @@ export const paraglide = createUnplugin((config: UserConfig) => {
 	let project: InlangProject | undefined = undefined
 	async function getProject(): Promise<InlangProject> {
 		if (project) return project
+
+		const repoRoot = await findRepoRoot({ nodeishFs: fs, path: projectPath })
+
+		const repo = repoRoot
+			? await openRepository(repoRoot, {
+					nodeishFs: fs,
+			  })
+			: await openRepository("file://", { nodeishFs: fs })
```

Comment:
not sure we should open the root, i would either fail without repo or try opening the cwd...

Reply:
Given that we want a repo to exists in the inlang SDK, opening a mockRepo should be the go-to solution no?

Reply:
@samuelstroschein  this is the implementation not the test, this nearly got me confused too...

Reply:
@janfjohannes now i am confused. wdym with implementation not the test? The file:// seems wrong. If no repo exists, `loadProject` should be called with `loadProject(mockRepo)`

Reply:
@LorisSigrist what wording do you find better `loadProject(virtualRepo())` or `loadProject(mockRepo())` in cases where no repo exists?

Reply:
If it's meant to be used in code `virtualRepo` seems more appropriate. `mockRepo` sounds like it's just for testing

Reply:
@janfjohannes i vote for `virtualRepo()` instead of `mockRepo()` given that we decided to build a hard dependency on repos in inlang, which requires the use of "virtual repos" where no repo is available. 

What do you think?

Reply:
I will add that `virtualRepo()` is not useable in paraglide in it's current form. It doesn't operate on the real file-system, but on a memfs. Calling `project.settings()` will not actually read the settings from a user's `project.inlang/settings.json` and instead error, because the file is not present in the virtual file-system.

For now the best solution if there is no repo is to open the `cwd` as if it were a repo. Git-operations will fail, but filesystem operations do work just fine. Given that we want lint rules to run in CI where a git repo may not be present, basic operations like reading the settings will always need to run without a real repo, so paraglide can rely on that continuing to work.

Reply:
@LorisSigrist  @samuelstroschein  hmm there is a big confusion about this feature.  mockRepo is ONLY  for testing with a mock repository that provides some conveneince to setup a memfs from an optional snapshot. virtual repo  as described here as a lix feauture to allow more features for opening folders without an existing repo has nothing to do with mockRepo but is a seperate thing. I am not sure it is a good idea or if we need this as many subtle things would work in unexpected ways or are hard to reason about eg. project ids would  randomly change etc.  currently if you open a folder in lix and it does not have a repo it will just fail all the function calls that require a repo (which is currently all except the nodeishFs  functions.) we could argue if its better to explicitly enable this behaviour or if openRepository should fail without a rpo present, but this is another discussion.

Reply:
@janfjohannes Yes, we need to think about a solution here that is _easily communicatable_. We have confusion here that we need to lift. 


My thought line is:

i wanted a virtualRepo for testing and production use cases. I don't see a difference between `mockRepo` and `virtualRepo`. A developer gets a "virtual repository" whether that is in memory or using files form the real fs doesn't seem to make a large difference? 

```ts
// in memory repository
const repo = await virtualRepo()

// for testing (in memory fs)
const repo = await virtualRepo({ fromSnapshot })

// in memory fs (for repo stuff) with real fs (the use case from @LorisSigrist) 
const repo = await virtualRepo({ usingFs: nodeFs })

``` 

we can also go for 

```ts
// tree-shakable imports
import * as virtualRepo from "@lix-js"

const repo = await virtualRepo.fromSnapshot()
const repo = await virtualRepo.inMemory()
const repo = await virtualRepo.withFs()
```

Reply:
we should have a sync call. not sure we are on the same page @samuelstroschein

Reply:
agree. setting a reminder for tmr

Reply:
This was also discussed in https://github.com/opral/monorepo/issues/1898 

After talking with @janfjohannes, I pushed a commit in PR #2060 to simplify the use of mockRepo() in loadProject. test.

Reply:
and opened a discussion in https://github.com/opral/monorepo/discussions/2061